.github/workflows/tests.yaml

```diff
@@ -250,6 +250,65 @@ jobs:
           path: target/release/mina
           retention-days: 7
 
+  unit-tests:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v5
+
+      - name: Setup build dependencies
+        uses: ./.github/actions/setup-build-deps
+
+      - name: Install libpcap
+        run: sudo apt-get update && sudo apt-get install -y libpcap-dev
+
+      - name: Use shared OCaml setting up steps
+        uses: ./.github/actions/setup-ocaml
+        with:
+          ocaml_version: ${{ env.OCAML_VERSION }}
+
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          toolchain: ${{ env.RUST_STABLE_VERSION }}
+          cache-prefix: unit-tests-${{ env.CACHE_VERSION }}
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Run unit tests
+        run: |
+          cargo nextest run --workspace --lib \
```

Comment:
Could we have a target in the Makefile for this instead, please? We try to have all the commands run by the CI in the Makefile.